src/analytics/selectContentTracker/selectContentTracker.ts

```diff
@@ -0,0 +1,95 @@
+import { BaseTracker } from "../baseTracker/baseTracker";
+import { SelectContentEventInterface } from "./selectContentTracker.interface";
+
+export class SelectContentTracker extends BaseTracker {
+  eventName: string = "select_content";
+  eventType: string = "event_data";
+
+  constructor() {
+    super();
+    this.initialiseEventListener();
+  }
+  /**
+   * Initialises the event listener for the document click event.
+   *
+   * @param {type} None
+   * @return {type} None
+   */
+  initialiseEventListener() {
+    const detailsElements = document.querySelectorAll("details");
+    detailsElements.forEach((detailsElement) => {
+      detailsElement.addEventListener(
+        "toggle",
+        this.trackSelectContent.bind(this),
+        false,
+      );
+    });
+  }
+
+  /**
+   * Tracks the form response and sends the data to the analytics platform.
+   *
+   * @param {Event} event - The toggle event triggered by clicking on details element.
+   * @return {boolean} Returns true if the event is successfully tracked, otherwise false.
+   */
+
+  trackSelectContent(event: Event): boolean {
+    if (!window.DI.analyticsGa4.cookie.consent) {
+      return false;
+    }
+
+    let element: HTMLDetailsElement = event.target as HTMLDetailsElement;
+
+    const SelectContentEvent: SelectContentEventInterface = {
+      event: this.eventType,
+      event_data: {
+        event_name: "select_content",
+        type: "details ui",
+        url: "undefined",
+        text: this.getSummaryText(element),
+        section: "undefined",
+        action: this.getActionValue(element),
+        external: "undefined",
+        link_domain: "undefined",
+        "link_path_parts.1": "undefined",
+        "link_path_parts.2": "undefined",
+        "link_path_parts.3": "undefined",
+        "link_path_parts.4": "undefined",
+        "link_path_parts.5": "undefined",
```

Comment:
These undefined pieces of data will need to be populated. Have a look at some of the other trackers and follow the same pattern. Any issues with a particular field, flag it to Sam :)

Reply:
Ohh ok but on the ticket and in the implementation guide, it says should be left undefined ?


https://govukverify.atlassian.net/wiki/spaces/PI/pages/3603791886/WIP+GA4+Implementation+Guide#When-should-this-fire%3F.9

https://govukverify.atlassian.net/jira/software/c/projects/DFC/boards/436?search=select%20content%20&selectedIssue=DFC-191

Reply:
@di-aholme I think we confirmed with Sam that this was the correct behaviour?